types/hapi__joi/index.d.ts

```diff
@@ -19,1378 +19,1883 @@
+        /**
+         * Allow relative URIs. Defaults to `false`.
+         */
+        allowRelative?: boolean;
+        /**
+         * Restrict only relative URIs. Defaults to `false`.
+         */
+        relativeOnly?: boolean;
+        /**
+         * Allows unencoded square brackets inside the query string.
+         * This is NOT RFC 3986 compliant but query strings like abc[]=123&abc[]=456 are very common these days.
+         * Defaults to `false`.
+         */
+        allowQuerySquareBrackets?: boolean;
+        /**
+         * Validate the domain component using the options specified in `string.domain()`.
+         */
+        domain?: boolean;
+    }
+
+    interface DataUriOptions {
+        /**
+         * optional parameter defaulting to true which will require = padding if true or make padding optional if false
+         */
+        paddingRequired?: boolean;
+    }
+
+    interface Base64Options {
+        /**
+         * optional parameter defaulting to true which will require = padding if true or make padding optional if false
+         */
+        paddingRequired?: boolean;
+
+        /**
+         * if true, uses the URI-safe base64 format which replaces + with - and \ with _. Defaults to false.
+         */
+        urlSafe?: boolean;
+    }
+
+    interface SwitchCases {
+        /**
+         * the required condition joi type.
+         */
+        is: SchemaLike;
+        /**
+         * the alternative schema type if the condition is true.
+         */
+        then: SchemaLike;
+    }
+
+    interface SwitchDefault {
+        /**
+         * the alternative schema type if no cases matched.
+         * Only one otherwise statement is allowed in switch as the last array item.
+         */
+        otherwise: SchemaLike;
+    }
+
+    interface WhenOptions {
+        /**
+         * the required condition joi type.
+         */
+        is?: SchemaLike;
+
+        /**
+         * the alternative schema type if the condition is true. Required if otherwise or switch are missing.
+         */
+        then?: SchemaLike;
+
+        /**
+         * the alternative schema type if the condition is false. Required if then or switch are missing.
+         */
+        otherwise?: SchemaLike;
+
+        /**
+         * the list of cases. Required if then is missing.  Required if then or otherwise are missing.
+         */
+        switch?: Array<SwitchCases | SwitchDefault>;
+
+        /**
+         * whether to stop applying further conditions if the condition is true.
+         */
+        break?: boolean;
+    }
+
+    interface WhenSchemaOptions {
+        /**
+         * the alternative schema type if the condition is true. Required if otherwise is missing.
+         */
+        then?: SchemaLike;
+        /**
+         * the alternative schema type if the condition is false. Required if then is missing.
+         */
+        otherwise?: SchemaLike;
+    }
+
+    interface Cache {
+        set(key: any, value: any): void;
```

Comment:
Can keys really be `any`?

Reply:
From [CacheInterface](https://hapi.dev/family/joi/?v=16.0.1#cache-interface) docs:

> Note that key and value can be anything including objects, array, etc.

I will add documentation to it.